backend/cfg/simplify_terminator.ml

```diff
@@ -98,6 +98,9 @@ let block (block : C.basic_block) =
       let l = Label.Set.min_elt labels in
       block.terminator <- { block.terminator with desc = Always l }
   | Switch labels -> simplify_switch block labels
-  | Call_no_return _ | Tailcall (Self _ | Func _) | Raise _ | Return -> ()
+  | Call_no_return _
+  | Tailcall (Self _ | Func _)
+  | Raise _ | Return | Poll_and_jump _ ->
```

Comment:
```suggestion
  | Raise _ | Return | Poll_and_jump _ ->
  (* CR xclerc: I wonder whether Merge_straightline_block should be updated
to optimize a Poll_and_jump block where block is empty, and
has a terminator such as Always block'.

(Note that the blocks are currently and correctly not merged
because Poll_and_jump block can raise. The optimization I
suggest would be to rewrite it to Poll_and_jump block'.) *)
```